main.tf

```diff
@@ -0,0 +1,91 @@
+locals {
+  ssm_instances = flatten([
+    for k, v in var.ecs_cluster_config : [
+      for instance in v.ssm_instances :
+      {
+        cluster_name        = k
+        instance_name       = instance
+        ssm_activation_tags = v.ssm_activation_tags
+      }
+    ]
+  ])
+  ssm_instances_distinct = { for instance in local.ssm_instances : "${instance.cluster_name}.${instance.instance_name}" => instance }
+}
+
+resource "aws_ecs_cluster" "default" {
+  for_each = var.ecs_cluster_config
+
+  name = each.key
+  tags = var.tags
+
+  setting {
+    name  = "containerInsights"
+    value = each.value.ecs_container_insights ? "enabled" : "disabled"
+  }
+
+  lifecycle {
+    ignore_changes = [
+      setting
```

Comment:
Make this optional, otherwise you can never change the containerInsights setting from code when the module is deployed.